File: 백준/Bronze/5086. 배수와 약수/배수와 약수.cc
#include <iostream>
using namespace std;

int sibal(int a, int b)
{
    if(a%b==0) return 1;
    else if(b%a==0) return 2;
    else return 3;
}

int main() {
    int a=1, b=1;
    while(1) {
        cin>>a>>b;
        if(a==0 && b==0) break;
        if(sibal(a, b)==1) cout<<"multiple"<<endl;
        else if(sibal(a, b)==2) cout<<"factor"<<endl;
        else if(sibal(a, b)==3) cout<<"neither"<<endl;
    }
    return 0;
}

File: 백준/Bronze/5086. 배수와 약수/배수와 약수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    while (true) {
      String inps[] = br.readLine().split(" ");
      int a = Integer.parseInt(inps[0]);
      int b = Integer.parseInt(inps[1]);
      if (a == 0 && b == 0) break;
      if (b % a == 0) bw.write("factor\n");
      else if (a % b == 0) bw.write("multiple\n");
      else bw.write("neither\n");
    }
    bw.flush();
  }
}
